controllers/replicas.go

```diff
@@ -552,30 +552,67 @@ func (r *ClusterReconciler) updateRoleLabelsOnPods(
 	return nil
 }
 
-// updateClusterRoleLabelsOnPVCs ensures that the PVCs have the correct cluster role label set
-func (r *ClusterReconciler) updateClusterRoleLabelsOnPVCs(
+// updateOperatorLabelsOnInstances ensures that the instances have the correct labels
+func (r *ClusterReconciler) updateOperatorLabelsOnInstances(
 	ctx context.Context,
-	pods corev1.PodList,
-	pvcs corev1.PersistentVolumeClaimList,
+	instances corev1.PodList,
 ) error {
-	for _, pod := range pods.Items {
-		podRole, podHasRole := pod.ObjectMeta.Labels[specs.ClusterRoleLabelName]
-		if !podHasRole {
+	for _, instance := range instances.Items {
+		instance := instance
```

Comment:
why are we creating a new variable here when we could just use the index?

instances.Items[i] would be a more efficient way of writing this